app/questionnaire/path_finder.py

```diff
@@ -13,45 +18,54 @@
 
 
 class PathFinder:
-    def __init__(self, schema, answer_store, metadata, completed_blocks):
+    def __init__(
+        self,
+        schema: QuestionnaireSchema,
+        answer_store: AnswerStore,
+        metadata: Mapping,
+        completed_store: CompletedStore = None,
+    ):
         self.answer_store = answer_store
         self.metadata = metadata
         self.schema = schema
-        self.completed_blocks = completed_blocks
-        self._full_routing_path = None
+        self.completed_store = completed_store
 
-    @staticmethod
-    def _block_index_for_location(blocks, location):
-        return next(
-            (
-                index
-                for (index, block) in enumerate(blocks)
-                if block['id'] == location.block_id
-            ),
-            None,
-        )
+    def is_path_complete(self, path):
+        return not self.get_first_incomplete_location_on_path(path)
 
-    def build_path(self):
-        """
-        Visits all the blocks from a location forwards and returns path
-        taken given a list of answers.
+    def get_first_incomplete_location_on_path(self, path):
```

Comment:
Could a type hint help this method name be less wordy? It's a collection of locations being passed in right?

Reply:
I don't think so as `path` is a well understood concept in runner. We could drop the `_on_path` bit so it becomes `get_first_incomplete_location`?

Reply:
Sounds good.